src/content/app/genome-browser/state/browser-location/browserLocationSlice.ts

```diff
@@ -0,0 +1,173 @@
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import { getChrLocationStr } from 'src/content/app/genome-browser/helpers/browserHelper';
+import { ParsedUrlPayload } from 'src/content/app/genome-browser/state/browser-entity/browserEntitySlice';
+import { buildFocusIdForUrl } from 'src/shared/state/ens-object/ensObjectHelpers';
+
+import { getChrLocation } from 'src/content/app/genome-browser/state/browser-location/browserLocationSelectors';
+import {
+  getBrowserActiveGenomeId,
+  getBrowserActiveEnsObjectId
+} from 'src/content/app/genome-browser/state/browser-entity/browserEntitySelectors';
+
+import { RootState } from 'src/store';
+
+const isServer = typeof window === 'undefined';
+const chrLocations = !isServer ? browserStorageService.getChrLocation() : [];
+
+export type ChrLocation = [string, number, number];
+
+export type ChrLocations = { [genomeId: string]: ChrLocation };
+
+export type BrowserLocationState = Readonly<{
+  chrLocations: ChrLocations; // final location of the browser when user stopped dragging/zooming; used to update the url
+  actualChrLocations: ChrLocations; // transient locations that change while user is dragging or zooming
+  regionEditorActive: boolean;
+  regionFieldActive: boolean;
+  isObjectInDefaultPosition: boolean;
+}>;
+
+export const defaultBrowserLocationState: BrowserLocationState = {
+  chrLocations,
+  actualChrLocations: {},
+  regionEditorActive: false,
+  regionFieldActive: false,
+  isObjectInDefaultPosition: false
+};
+
+export const setActualChrLocation: ActionCreator<
+  ThunkAction<any, any, null, Action<string>>
+> = (chrLocation: ChrLocation) => {
+  return (dispatch: Dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (!activeGenomeId) {
+      return;
+    }
+    const payload = {
+      [activeGenomeId]: chrLocation
+    };
+
+    dispatch(updateActualChrLocation(payload));
+  };
+};
+
+export const setChrLocation: ActionCreator<
+  ThunkAction<any, any, null, Action<string>>
+> = (chrLocation: ChrLocation) => {
+  return (dispatch: Dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    const activeEnsObjectId = getBrowserActiveEnsObjectId(state);
+    const savedChrLocation = getChrLocation(state);
+    if (!activeGenomeId || !activeEnsObjectId) {
+      return;
+    }
+    const payload = {
+      [activeGenomeId]: chrLocation
+    };
+
+    dispatch(updateChrLocation(payload));
+    browserStorageService.updateChrLocation(payload);
+
+    if (!isEqual(chrLocation, savedChrLocation)) {
+      const newUrl = urlFor.browser({
+        genomeId: activeGenomeId,
+        focus: buildFocusIdForUrl(activeEnsObjectId),
+        location: getChrLocationStr(chrLocation)
+      });
+      dispatch(replace(newUrl));
+    }
```

Comment:
Is it possible to remove this block of code relying on connecter-react-router? Is it even necessary?

Reply:
I tried it but it breaks the url location update.